lib/puppet/util/windows/adsi.rb

```diff
@@ -298,13 +299,12 @@ def add_flag(flag_name, value)
     end
 
     def password=(password)
-      if !password.nil?
+      unless password.nil?
         native_object.SetPassword(password)
-        commit
       end
 
-      fADS_UF_DONT_EXPIRE_PASSWD = 0x10000
-      add_flag("UserFlags", fADS_UF_DONT_EXPIRE_PASSWD)
+      # commit is idempotent
+      commit
```

Comment:
The original code would not commit the changes for a `nil` password; however if called from `create`, the user would still be committed. The code here commits for a `nil` password as well. Is this ok? Note that because we did not commit for a `nil` password, `add_flag` had a [`rescue 0` clause](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/adsi.rb#L293) since if this was called from `create`, retrieving the user flag would raise an error because the user has not yet been committed on the system. I don't think swallowing errors is a good idea, which is also why I am committing even under a nil password. Is this ok?